optimizer/evaluator/evaluator.go

```diff
@@ -973,3 +975,14 @@ func (e *Evaluator) evalAggSetValue(v *ast.AggregateFuncExpr) {
 	ctx := v.GetContext()
 	v.SetValue(ctx.Value)
 }
+
+func (e *Evaluator) evalAggAvg(v *ast.AggregateFuncExpr) {
+	ctx := v.GetContext()
+	switch x := ctx.Value.(type) {
+	case float64:
+		ctx.Value = x / float64(ctx.Count)
```

Comment:
If Count is zero, ctx.Value must be nil. So it will not be in this switch branch. I will add a test case for it.